lib/propmangle.js

```diff
@@ -167,9 +167,6 @@ function mangle_properties(ast, options) {
     var private_cache = new Map();
     if (options.cache) {
         cache = options.cache.props;
-        cache.forEach(function(mangled_name) {
-            reserved.add(mangled_name);
-        });
```

Comment:
I think we should instead be pushing to `unmangleable`. It fulfills the same role as `reserved`, but is only checked when `nth_identifier` comes up with a new mangled name.

The reason this was added was to stop mangling from creating colliding names, eg. if something mangles to X but there's already an X in a previous file that Terser mangled.

Here's when this reserved stuff was added, for more context:

https://github.com/mishoo/UglifyJS/pull/2722/files#diff-4f3203fe6e0c32296500192f4d6ffc6c6015ff3596b7bb5b46217a1b06ef10e6R115-R120

Reply:
Would've added a fancy GH suggestion, but `unmangleable` is defined after this so it's not as simple as that.

Reply:
Good catch. Fixed, and updated the test for this case.